File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/image/ImageCreateDto.java
package com.intexsoft.webshop.productservice.dto.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ImageCreateDto {
    @NotBlank
    @Size(max = 512)
    String filePath;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/image/ImageCreateDto.kt
package com.intexsoft.webshop.productservicekt.dto.image

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
@JsonIgnoreProperties(ignoreUnknown = true)
data class ImageCreateDto(
    @field:NotBlank
    @field:Size(max = 512)
    val filePath: String?
)